test/unit/sql/add-constraint.test.js

```diff
@@ -68,7 +68,10 @@ if (current.dialect.supports.constraints.addConstraint) {
               }
             }
           }), {
-            default: 'ALTER TABLE [myTable] ADD CONSTRAINT [check_mycolumn_where] CHECK (([myColumn] > 50 AND [myColumn] < 100));'
+            default: [
```

Comment:
I'm not really sure why it changes or if it is caused by the code changes. 99% of the time I'm getting the original result but some time they get flipped.
In general JS asks you not to rely on the order when traversing object properties. I never actually had a case where they weren't traverse in the same order but I'm not sure if I ever had a code where it would matter.

Reply:
It make testing things a bit hard, may be we should sort object keys to have single defined output for same arguments

Reply:
I can look into it

Reply:
@yonjah Did you figure out why this is an issue, it would be great if we can get some deterministic behavior here

Reply:
From what I could see I only get random results on node v4 and it is due to the fact that the alias method delete the previous keys -
https://github.com/yonjah/sequelize/blob/fb8188efb78eb21f91f87cbe0693220adbdf7ee9/lib/dialects/abstract/query-generator.js#L2107

This is how the old aliasing worked but I can change the alias logic to generate a new object instead of deleting properties on the old one (which might be a better solution any way but a bit more complex)